src/main/java/org/springframework/data/hazelcast/HazelcastKeyValueAdapter.java

```diff
@@ -38,8 +38,15 @@
 
 	private HazelcastInstance hzInstance;
 
+    /**
+     * Creates an instance of {@link HazelcastKeyValueAdapter}.
+     *
+     * @deprecated Use {@code new HazelcastKeyValueAdapter(Hazelcast.getOrCreateHazelcastInstance(new Config(Constants
+     * .HAZELCAST_INSTANCE_NAME))} instead. This method will be removed in a future release.
+     */
+	@Deprecated
 	public HazelcastKeyValueAdapter() {
-		this(Hazelcast.getOrCreateHazelcastInstance(new Config(Constants.HAZELCAST_INSTANCE_NAME)));
+	    this(Hazelcast.getOrCreateHazelcastInstance(new Config(Constants.HAZELCAST_INSTANCE_NAME)));
```

Comment:
The core issue was that the Query mechanism used by default an instance with the name `Constants.HAZELCAST_INSTANCE_NAME`. So if it didn't exist (like in the sample from README (!) ), then it failed with NPE. 

Now, when user defines his instance (like in the README), then this instance is used. Otherwise, the instance with the name `Constants.HAZELCAST_INSTANCE_NAME` is used.

I wouldn't add any additional mechanism (like looking if any instance exist with `getAllHazelcastInstances() ), since then we 1) Make it more complex 2) Make it non-backwards compatible, 3) As you mentioned, we will remove it anyway.